GAD 8 - Binomial Heap/tests/binomilia/BinomialHeapTester.java

```diff
@@ -0,0 +1,136 @@
+package gad.tests;
+
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import gad.binomilia.*;
+
+/**
+ * 
+ * @author Mihir Mahajan
+ * 
+ *         These tests only test if your BinomialHeap/BinomialTree itself works.
+ *         They don't test the logging. Still, they may not work if the logging
+ *         is incorrect!
+ * 
+ *         I can not guarantee the correctness or accuracy of these tests. I
+ *         hope they help, enjoy!
+ */
+
+public class BinomialHeapTester {
+
+	private BinomialHeap heap;
+	private StudentResult result;
+
+	@BeforeEach
+	void init() {
+
+		heap = new BinomialHeap();
+		result = new StudentResult();
+	}
+
+	@Test
+	void testIllegal() {
+		assertThrows(RuntimeException.class, () -> heap.min(),
+				"Please throw an exception when heap is empty, since there cant be a minimal element in an empty heap. (NoSuchElementExeption would be cool)");
+		assertThrows(RuntimeException.class, () -> heap.deleteMin(result),
+				"Please throw an exception when heap is empty, since deleteMin() is illegal on an empty heap. (NoSuchElementExeption would be cool)");
+	}
+
+	@Test
+	void testMin() {
+		for (int i = 0; i < 500; i++) {
+			heap.insert(i, result);
+			assertEquals(-300, heap.min(), "Wrong value for min!");
```

Comment:
This will not work when you start the loop at 0. I don't have an issue with negative loop starts if you actually use those values. Right now the min element in the heap is 0